cranker/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+                            now,
+                            expiry_time
+                        );
+                        match self.claim_pool_tokens(&receipt).await {
+                            Ok(_) => {
+                                info!("Successfully claimed tokens for receipt {}", receipt.base);
+                                let mut metrics = self.metrics.lock().unwrap();
+                                metrics.successful_claims += 1;
+                            }
+                            Err(e) => {
+                                error!(
+                                    "Failed to claim tokens for receipt {}: {}",
+                                    receipt.base,
+                                    e
+                                );
+                                let mut metrics = self.metrics.lock().unwrap();
+                                metrics.failed_claims += 1;
+                            }
+                        }
+                    } else {
+                        info!(
+                            "Receipt {} not yet expired. Current time: {}, Expiry time: {}",
+                            receipt.base,
+                            now,
+                            expiry_time
+                        );
+                    }
+                }
+                None => {
+                    error!(
+                        "Receipt {} has invalid timing values - would overflow. Deposit time: {}, Cool down: {}",
+                        receipt.base,
+                        deposit_time,
+                        cool_down
+                    );
+                }
+            }
+        }
+        Ok(())
+    }
+
+    async fn get_deposit_receipts(&self) -> Result<Vec<DepositReceipt>, CrankerError> {
+        let discriminator = StakeDepositInterceptorDiscriminators::DepositReceipt as u8;
+        info!("Searching for deposit receipts");
+
+        let filters = vec![
+            RpcFilterType::Memcmp(
+                Memcmp::new_base58_encoded(
+                    0, // offset
+                    &[discriminator] // data to match
+                )
+            )
+        ];
+
+        let accounts = self.rpc_client
+            .get_program_accounts_with_config(&self.program_id, RpcProgramAccountsConfig {
+                filters: Some(filters),
+                account_config: RpcAccountInfoConfig {
+                    encoding: Some(solana_account_decoder::UiAccountEncoding::Base64),
+                    commitment: Some(CommitmentConfig::confirmed()),
+                    ..Default::default()
+                },
+                ..Default::default()
+            })
+            .map_err(CrankerError::RpcError)?;
+
+        info!("Found {} raw accounts", accounts.len());
+
+        Ok(
+            accounts
+                .into_iter()
+                .filter_map(|(pubkey, account)| {
+                    // Skip the 8-byte discriminator
```

Comment:
Could probably use `DepositReceipt::try_from_slice_unchecked(account.data.as_slice())` to shorten this up. Just need to add `use jito_bytemuck::AccountDeserialize;` to the top.